independent-projects/tools/devtools-testing/src/main/java/io/quarkus/devtools/testing/codestarts/QuarkusCodestartTestBuilder.java

```diff
@@ -209,14 +209,12 @@ public QuarkusCodestartTestBuilder standaloneExtensionCatalog(String quarkusBomG
             Objects.requireNonNull(buildWithQuarkusCoreVersion, "quarkus version not found in extensions");
             String quarkusVersion = quarkusBomVersion != null ? quarkusBomVersion : buildWithQuarkusCoreVersion;
             enableRegistryClientTestConfig(quarkusBomGroupId != null ? quarkusBomGroupId : "io.quarkus", quarkusVersion);
-            final ExtensionCatalog extensionCatalog = QuarkusProjectHelper
-                    .getExtensionCatalog(quarkusVersion);
+            final ExtensionCatalog extensionCatalog = QuarkusProjectHelper.getExtensionCatalog(quarkusVersion);
             disableRegistryClientTestConfig();
-            if (!(extensionCatalog instanceof ExtensionCatalog.Mutable)) {
-                throw new IllegalStateException("Problem with the given ExtensionCatalog type");
-            }
-            extensions.addAll(extensionCatalog.getExtensions());
-            ((ExtensionCatalog.Mutable) extensionCatalog).setExtensions(extensions);
+            final ExtensionCatalog.Mutable mutableCatalog = extensionCatalog instanceof ExtensionCatalog.Mutable
+                    ? (ExtensionCatalog.Mutable) extensionCatalog
+                    : extensionCatalog.mutable();
+            extensions.forEach(mutableCatalog::addExtension);
             this.extensionCatalog = extensionCatalog;
```

Comment:
Should we be using a built catalog in the event we had to change it / add extensions?